builtins/amp-ad.js

```diff
@@ -263,6 +263,64 @@ export function installAd(win) {
         this.element.removeChild(this.iframe_);
       });
     }
+
+    /**
+     * Takes an AMP ad and immediately renders.
+     * @param {string} ampHtmlAd AMP HTML that will be immediately rendered if
+     *   it is validated.
+     * @private
+     */
+    validateAndRenderOrFallback_(ampHtmlAd) {
+      // First find the body of the valid AMP document. Wrap this in an
+      // identifiable div and append it as a child to our amp-ad tag.
+      var ampBodyStart = ampHtmlAd.indexOf("<body>");
+      if(ampBodyStart > 0) {
+        var ampBodyEnd = ampHtmlAd.lastIndexOf("</body>");
+        var div = createDivContainer(this.element.ownerDocument.defaultView,
+            this.element);
+        var uniqueId = " #" + div.id + " ";
+        div.innerHTML = ampHtmlAd.substring(
+          ampBodyStart + "<body>".length, ampBodyEnd);;
+        this.element.appendChild(div);
+
+
+        // Next append any styles to our amp-custom style tag. Use the unique
+        // div id to nest these styles appropriately.
+        var ampStyleStart = ampHtmlAd.indexOf("<style amp-custom>");
+        if(ampStyleStart > 0) {
+          var ampStyleEnd = ampHtmlAd.indexOf("</style>", ampStyleStart);
+          var ampStyle = ampHtmlAd.substring(ampStyleStart +
+            "<style amp-custom>".length, ampStyleEnd);
+
+          // Split the string on ending braces, remove the empty item at the end,
+          // and create new set of styles nesting the existing styles as
+          // descendants of the unique id of the div.
+          var splitArray = ampStyle.split('}');
+          splitArray.pop();
+          ampStyle = uniqueId + splitArray.join('} ' + uniqueId) + '}');
+
+          // Lastly find the amp-custom tag and append to that.
+          var styleTags = document.getElementsByTagName("style");
+          var customStyleTag = null;
+          for (var i = 0; i < styleTags.length; i++)
```

Comment:
AMP only allows 1 style tag, right?  If so, you could do a query selector:  document.querySelector('style[amp-custom]')